shard/src/ext/purge_command.py

```diff
@@ -1,41 +1,192 @@
+        await ctx.send("Need to use command with id or time parameter")
+        return
+
+
+@purge.command()
+async def id(ctx, mid, inclusive=False):
+    """
+    Purge a channel of messages until a specific message is reached
+
+    Usage: {prefix}purge id {message id} [true]
+    Purges all messages sent after the message corresponding to the
+    message id of the user who sent that message. If the optional last
+    parameter is set to true or True then deleted messages will not be
+    limited to just the one author but included all messages in the relevant
+    time period. Any non "true" or "True" value passed in at the end will
+    default to False.
+
+    Only searches the past 500 messages sent in a channel for which to delete.
+    If more than 500 messages have been sent between the given message's id and
+    when this command is run, then all of that user's messages may not get
+    deleted.
+
+    NOTE: If the original message is deleted before the bot deletes it, all messages
+    from the target message's author in the channel will be deleted depending on the
+    timing of when the original message was deleted.
+    """
+    message_id = None
+    try:
+        message_id = int(mid)
+    except ValueError:
+        await ctx.send("Message ID has an invalid format")
+        return
+
     settings = ctx.bot.settings[ctx.guild]
-    user = ctx.bot.user
-
-    for arg in args:
-        with suppress(ValueError):
-            user = ctx.guild.get_member(int(arg))
-            if user:
-                break
-    else:
-        user = ctx.bot.user
-
-    for arg in args:
-        with suppress(ValueError):
-            if int(arg) < 100000:
-                count = int(arg)
-    else:
-        count = 100
-
-    channel = ctx.channel
-    if ctx.message.channel_mentions:
-        channel = ctx.message.channel_mentions[0]
-
-    async with ctx.channel.typing():
-        if (ctx.author.id in settings.admins_ids):
-            deleted = await channel.purge(limit=count, check=lambda m: m.author == user)
-            await ctx.send('Deleted {} message(s)'.format(len(deleted)))
+    if ctx.author.id not in settings.admins_ids:
+        await ctx.send("You do not have permissions to purge messaages")
+        return
+
+    try:
+        original_message = await ctx.channel.fetch_message(message_id)
+    except discord.NotFound:
+        await ctx.send("Original message has already been deleted. "
+                       "Try again with different message.")
+        return
+    except discord.Forbidden:
+        await ctx.send("Architus does not have permission to read message history")
+        return
+    except discord.HTTPException:
+        await ctx.send("Failed to reach discord servers. Try again in a bit")
+        return
+    except Exception:
+        await ctx.send("Something went wrong. :(")
+        return
+
+    messages_to_delete = []
+    async for m in ctx.channel.history(limit=500):
```

Comment:
this should probably be higher, like maybe 5 or 10k, depending on how slow discord is

Reply:
Okay. I wasn't entirely sure what the limit should be